src/com/facebook/buck/android/MergeAndroidResourcesStep.java

```diff
@@ -152,15 +157,30 @@ public static MergeAndroidResourcesStep createStepForUberRDotJava(
         overrideSymbolsPath,
         unionPackage,
         /* rName */ Optional.empty(),
-        false);
+        /* useOldStyleableFormat */ false,
+        /* skipPrebuiltRDotJava */ false);
   }
 
   public ImmutableSortedSet<Path> getRDotJavaFiles() {
-    return FluentIterable.from(androidResourceDeps)
-        .transform(HasAndroidResourceDeps::getRDotJavaPackage)
-        .append(unionPackage.map(Collections::singletonList).orElse(Collections.emptyList()))
-        .transform(this::getPathToRDotJava)
-        .toSortedSet(natural());
+    if (skipPrebuiltRDotJava) {
+      return
+          FluentIterable.from(unionPackage
+              .map(Collections::singletonList)
+              .orElse(Collections.emptyList())
+          )
+          .transform(this::getPathToRDotJava)
+          .toSortedSet(natural());
+
+    } else {
+      return FluentIterable.from(androidResourceDeps)
+          .transform(HasAndroidResourceDeps::getRDotJavaPackage)
+          .append(unionPackage
+              .map(Collections::singletonList)
+              .orElse(Collections.emptyList())
+          )
+          .transform(this::getPathToRDotJava)
```

Comment:
Rather than duplicating code between the branches, do something like the below. It makes the function of the switch clearer and eliminates the possibility of drift in the duplicated code over time.
```
FluentIterable<String> packages = FluentIterable.from(
    unionPackage.map(Collections::singletonList).orElse(Collections.emptyList());

if (!skipPrebuiltRDotJava) {
    packages = packages.append(
        FluentIterable.from(androidResourceDeps)
            .transform(HasAndroidResourceDeps::getRDotJavaPackage));
}

return packages
    .transform(this::getPathToRDotJava)
    .toSortedSet(natural());
```

Reply:
good point, removed duplicate as you suggested.